compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -1020,6 +1020,14 @@ fn report_non_exhaustive_match<'p, 'tcx>(
 
     let mut suggestion = None;
     let sm = cx.tcx.sess.source_map();
+    let suggested_arm = if witnesses.len() < 4
```

Comment:
why < 4?

Reply:
Because that's also the limit used above to choose whether to report patterns or `_`. You're right this should be made clear, let me clean this

Reply:
I just find it strange that this applies to only the never branch, since the other half of the `if` could suffer from the same problem?

Reply:
There were actually three places using this magic threshold ^^' Fixed in the last commit

Reply:
If there are more than 4 witnesses we suggest `_ => todo!()`, it doesn't make sense to remove the `todo!()` here. And I can't just suggest `!` either because the type might not be empty.